core/src/main/java/io/smallrye/context/SmallRyeContextManager.java

```diff
@@ -246,15 +245,13 @@ public SmallRyeThreadContext allClearedThreadContext() {
         private ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
         private boolean addDiscoveredThreadContextProviders;
         private boolean addDiscoveredContextManagerExtensions;
-        private List<ThreadContextProvider> contextProviders = new ArrayList<>();
-        private List<ContextManagerExtension> contextManagerExtensions = new ArrayList<>();
+        private final List<ThreadContextProvider> contextProviders = new ArrayList<>();
+        private final List<ContextManagerExtension> contextManagerExtensions = new ArrayList<>();
         private ExecutorService defaultExecutorService;
 
         @Override
         public Builder withThreadContextProviders(ThreadContextProvider... providers) {
-            for (ThreadContextProvider contextProvider : providers) {
-                contextProviders.add(contextProvider);
-            }
+            contextProviders.addAll(Arrays.asList(providers));
```

Comment:
Are you sure this is better? It creates an intermediary list…

Reply:
Oh right. It's definitely not better. I'll remove it

Reply:
Fixed